sky/execution.py

```
@@ -314,6 +315,38 @@ def _execute(
             logger.info('Dryrun finished.')
             return None, None
 
+        # TODO(andyl): extend to more clouds beyond Kubernetes.
+        high_availability_specified = False
+        controller = controller_utils.Controllers.from_name(cluster_name)
+        if controller is not None:
```

we should raise this error before the controller is launched.

---

even before the confirmation in CLI.

---

Some errors need to check `launched_resources` cannot be moved up.